sinks.go

```diff
@@ -46,6 +46,32 @@ func (p *Pipe) CountLines() (int, error) {
 	return lines, p.Error()
 }
 
+// Slice return each lines of the pipe in a Slice or an error.
+// If there is an error reading the pipe, the pipe's error status
+// is also set.
+func (p *Pipe) Slice() ([]string, error) {
+	if p == nil || p.Error() != nil {
+		return nil, p.Error()
+	}
+	value, err := p.String()
```

Comment:
It's a shame to allocate memory for the entire contents of the pipe as a string, and then throw it away, isn't it? Maybe it's a better idea to read the pipe using `EachLine()`, and build up the slice of strings that way?

Reply:
That's what I did in my first version : https://github.com/bitfield/script/blob/f74894fd9e143fecf4195bd266b091e766370086/sinks.go#L57 
but you had this [feedback](https://github.com/bitfield/script/pull/47/commits/f74894fd9e143fecf4195bd266b091e766370086#r407059026) that's why I end up with this solution.